department-of-veterans-affairs/caseflow
# Hide mandate and judgement date for MDR

Resolves [CASEFLOW-96](https://vajira.max.gov/browse/CASEFLOW-96) and the frontend of [CASEFLOW-282](https://vajira.max.gov/browse/CASEFLOW-282).

### Description
When a user is inputting a CAVC remand, if they select that the remand is an MDR (Memorandum Decision on Remand), the fields for judgement and mandate date should disappear, because they are inapplicable.

### Acceptance Criteria
#### For CASEFLOW-96
- [x] <strike>Please put this work behind the feature toggle: mdr_cavc_remand</strike> (implicitly met: the MDR option doesn't appear unless that feature toggle is on)
- [x] This feature should be accessible to the following user groups: CAVC Litigation Support
- [x] Include screenshot(s) in the Github issue if there are front-end changes
- [x] <strike>Update documentation: [CAVC Remand](https://github.com/department-of-veterans-affairs/caseflow/wiki/CAVC-Remands)</strike> It already addresses that mandate and judgement dates aren't available.
- [x] Verify when the MDR radio button is selected on the CAVC entry form the judgement & mandate date is disabled

#### For CASEFLOW-282
- [x] <strike>Please put this work behind the feature toggle: dd_cavc_remand</strike> (implicitly met; also N.B. that the feature toggle is `dismissal_cavc_remand`)
- [x] This feature should be accessible to the following user groups: CAVC Litigation Support Team
- [x] Update documentation: CAVC Remand (still nothing to update there)
- [x] Death Dismissal is a radio button on the CAVC entry form
- [ ] <strike>The following sub-text displays under the Death Dismissal radio button: see `radio-sub-text`</strike>
        requirement moved to [CASEFLOW-915](https://vajira.max.gov/browse/CASEFLOW-915)
- [x] If the Death Dismissal radio button and the No radio button under the "are judgement and mandate dates provided" question then the following copy should appear: see `no-mandate-text`
- [x] If the Yes radio button is clicked under the "Are judgement and mandate dates provided?" question then the following date picker fields display:
    What is the Court's judgement date? 
    What is the Court's mandate date?

Display text:
`radio-sub-text` "e.g. if the veteran has passed away with no substitution"
`no-mandate-text` "This task will be put on hold for 90-days to wait for the Court's judgement and mandate dates. This case will be accessible in the On Hold tab and will return to the team's Unassigned tab when the 90-days has passed.


### Testing Plan
1. Deploy this branch and log in as a CAVC Lit Support user
2. Ensure the `mdr_cavc_Remand` feature toggle is on (it probably is already)
3. Create an appeal ready for a CAVC remand (this code shamelessly stolen from [Yoom's PR](https://github.com/department-of-veterans-affairs/caseflow/pull/15820))
```ruby
original_appeal = FactoryBot.create(:appeal, 
                           :dispatched, 
                           associated_judge: JudgeTeam.first.judge, 
                           associated_attorney: JudgeTeam.first.attorneys.first);
FactoryBot.create_list(:request_issue, 2,                                                                                  
            :rating,                                                                                            
            :with_rating_decision_issue,                                                                           
            decision_review: original_appeal,                                                                               
            veteran_participant_id: Veteran.first.participant_id,                                                        
            contested_issue_description: "issue_description",                                                              
            notes: "issue_notes");

original_appeal.uuid
```
4. Visit the appeal with that UUID (`/queue/appeals/UUID`) and press "+ Add CAVC Remand"
5. Confirm that selecting MDR as the remand subtype (Remand for "How are you proceeding?" and MDR for "What type of remand is this?") causes the judgement and mandate date fields to disappear. (The blue info box is desired but not added by this PR.)
6. Futz around with the form a bit to make sure it doesn't break as you choose different options
7. Fill it out with MDR selected and ensure it is created successfully

### User Facing Changes

#### Before
<img width="502" alt="Screen Shot 2021-01-28 at 4 09 32 PM" src="https://user-images.githubusercontent.com/314663/106201569-a1067900-6186-11eb-9049-3b84013a0708.png">

(Note the three date fields)

#### After
<img width="1208" alt="Screen Shot 2021-01-28 at 4 11 46 PM" src="https://user-images.githubusercontent.com/314663/106201603-afed2b80-6186-11eb-8243-5255bc90117c.png">

Note that only decision date is available. (Blue info box was pre-existing but explains why the fields are absent.)


### Storybook Story
Interestingly (to me), this one came for free.

<img width="1444" alt="Screen Shot 2021-01-28 at 4 21 43 PM" src="https://user-images.githubusercontent.com/314663/106200569-2e48ce00-6185-11eb-81ba-77dffda0957c.png">

I think it's because the Storybook code has this:

```
  return <Wrapper {...storeArgs}>
    <AddCavcRemandView appealId={appealId} {...componentArgs} />
  </Wrapper>;
```
